subprojects/model-core/src/main/java/org/gradle/api/internal/model/InstantiatorBackedObjectFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.model;
+
+import org.gradle.api.Named;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.reflect.ObjectInstantiationException;
+import org.gradle.internal.reflect.Instantiator;
+
+public class InstantiatorBackedObjectFactory implements ObjectFactory {
```

Comment:
:o: I don't find strongly about it but I'd probably would have named the class something with `Adapter` at the end.